fahrschein-spring-boot-starter/src/main/java/org/zalando/spring/boot/fahrschein/nakadi/config/FahrscheinRegistrar.java

```diff
@@ -65,15 +66,16 @@ private String registerPublisher(PublisherConfig publisherConfig) {
         });
     }
 
-    private String registerNakadiClient(AbstractConfig consumerConfig, String type) {
-        return registry.registerIfAbsent(consumerConfig.getId() + "-" + type, NakadiClient.class, () -> {
-            log.info(LOG_PREFIX + "NakadiClient ...", consumerConfig.getId());
-            final String requestFactoryRef = registerRequestFactory(consumerConfig);
+    private String registerNakadiClient(AbstractConfig config, String type) {
+        return registry.registerIfAbsent(config.getId() + "-" + type, NakadiClient.class, () -> {
+            log.info(LOG_PREFIX + "NakadiClient ...", config.getId());
+            final String requestFactoryRef = registerRequestFactory(config);
             return genericBeanDefinition(FahrscheinNakadiClientFactory.class)
-                .addConstructorArgValue(consumerConfig)
-                .addConstructorArgReference(registerCursorManager(consumerConfig, requestFactoryRef))
-                .addConstructorArgReference(registerObjectMapper(consumerConfig))
+                .addConstructorArgValue(config)
+                .addConstructorArgReference(registerCursorManager(config, requestFactoryRef))
+                .addConstructorArgReference(registerObjectMapper(config))
                 .addConstructorArgReference(requestFactoryRef)
+                .addConstructorArgValue(registry.getBeansOfType(EventPublishingHandler.class).values().stream().toList())
```

Comment:
`Picks up all PublishingHandlers and adds them to the NakadiClient`
I just saw the above point in the PR description.

I would suggest to load only the required handlers specified by the client via config file + `InstrumentedPublishingHandler`, instead of loading all handler implementations.   This would keep the library more flexible and also will allow clients to use specific handlers for specific publishers.

Reply:
IMO that would mean optimizing for some weird edge case, assuming that a) the client implements its own `PublishingHandler`, and b) registers multiple publishers which would require a different set of `PublishingHandlers`.

The current code would "just work" for the common-sense setup, even on custom `PublishingHandler` implementations, which is IMO the assumption for Spring Boot Autostarters.

How would you envision a client registering its own `PublishingHandler` in your approach? They would still have to create their own `NakadiClient` bean, and do all the wiring by hand, right? So, there is technically nothing gained by not pulling all `PublishingHandlers` by default.

Reply:
By default, loading the `InstrumentedPublishingHandler` in the presence of a `Tracer` makes sense, since we are the owners of this implementation. It offers the client a hassle-free, commonly-used setup without requiring manual configuration.

However, if we automatically load all other custom handlers without the client's consent, we risk introducing side effects, especially in scenarios involving multiple publishers (though not typical, still possible). Additionally, maintaining the execution order of handlers would become non-deterministic in such cases.

> Clients would still need to create their NakadiClient bean and perform the necessary wiring manually.

To address this, we can provide clients with a provision to specify handler beans in the configuration file, as demonstrated below:

```
fahrschein:
  publishers:
     handlers:
          - handlerOne
          - handlerTwo
```

This allows clients to define the handlers they want to use.

IMHO, the decision to load all request handlers can be deferred for now, allowing us to further refine the use case. Additionally, we need to undertake some refactoring in the existing code to ensure it is free from side effects before introducing this feature.

Reply:
hm, ok. Will push an update to the PR.